session.go

```diff
@@ -14,6 +14,100 @@ import (
 	"github.com/yawning/bulb"
 )
 
+// MortalListener can be killed at any time.
```

Comment:
I wonder... it should be possible to abstract away the listener semantics of the listener and make the API more generalized to handle the "polling a blocking call" pattern... then I could also use this same pattern elsewhere where blocking calls are used... say for example when capturing packets.